server/src/main/java/org/elasticsearch/snapshots/SnapshotShardsService.java

```diff
@@ -249,7 +257,13 @@ private void handleUpdatedSnapshotsInProgressEntry(String localNodeId, Snapshots
                         // due to CS batching we might have missed the INIT state and straight went into ABORTED
```

Comment:
Good point, but I think it makes more sense to handle this in `SnapshotsService` since the update which moves it to `WAITING` might already be in flight as well. If the whole snapshot is `ABORTED` then it doesn't make sense to have it have any `WAITING` shards, we should move them straight to `FAILED`.